ios/RCTWebRTC/WebRTCModule+DevicesManager.m

```diff
@@ -0,0 +1,301 @@
+//
+//  WebRTCModule+DevicesManager.m
+//  react-native-webrtc
+//
+//  Created by Filipi Fuchter on 08/03/22.
+//
+
+#import "WebRTCModule.h"
+
+@interface WebRTCModule (DevicesManager)
+
+@end
+
+@implementation WebRTCModule (DevicesManager)
+
+static NSString const *DEVICE_KIND_VIDEO_INPUT = @"videoinput";
+static NSString const *DEVICE_KIND_AUDIO_INPUT = @"audioinput";
+static NSString const *DEVICE_KIND_AUDIO_OUTPUT = @"audiooutput";
+
+enum DeviceType {
+    BLUETOOTH=1,
+    SPEAKER=2,
+    EARPIECE_HEADSET=3,
+    BUILT_IN_MICROPHONE=4,
+    BUILT_IN_MICROPHONE_SPEAKER=5,
+    CAMERA_USER=6,
+    CAMERA_ENVIRONMENT=7,
+};
+
+enum AudioRoute {
+    ROUTE_BUILT_IN=1,
+    ROUTE_SPEAKER=2,
+    ROUTE_BLUETOOTH=3,
+};
+
+- (NSNumber*)getAudioRouteFromDeviceType:(enum DeviceType)deviceType {
+    switch (deviceType) {
+        case EARPIECE_HEADSET:
+        case BUILT_IN_MICROPHONE:
+            return [NSNumber numberWithInt:ROUTE_BUILT_IN];
+        case BUILT_IN_MICROPHONE_SPEAKER:
+        case SPEAKER:
+            return [NSNumber numberWithInt:ROUTE_SPEAKER];
+        case BLUETOOTH:
+            return [NSNumber numberWithInt:ROUTE_BLUETOOTH];
+        default:
+            return 0;
+    }
+}
+
+RCT_EXPORT_METHOD(enumerateDevices:(RCTResponseSenderBlock)callback)
+{
+    NSLog(@"[Daily] enumerateDevice from DevicesManager");
+    NSMutableArray *devices = [NSMutableArray array];
+    
+    [self fillVideoInputDevices:devices];
+    [self fillAudioInputDevices:devices];
+    [self fillAudioOutputDevices:devices];
+    
+    callback(@[devices]);
+}
+
+// Whenever any headphones plugged in, it becomes the default audio route even if there is also bluetooth device.
```

Comment:
ah so this is a behavioral difference from android?

Reply:
wait...why is this comment above a method related to video devices? 😛

Reply:
> so...if it's not actually possible to use a bluetooth device if a wired headset is plugged in, should we remove the bluetooth headset from the list of input devices returned by `enumerateDevices()` if a wired headset is plugged in?

this might be more reason to make sure we show appropriate errors when the customer tries to pick an audio route that can't be fulfilled (or will have to behave differently than they intend)!

Reply:
Hi, yes, this behavior is different from Android. On Android we can still choose to use the Bluetooth, even if the wired headset is plugged.

And yes, if the user explicit chooses the audio route "bluetooth" when a bluetooth device is available, it'll still use the wired headset in this case. I have just tested again that.

I like your suggestion, of removing the bluetooth headset from the list of input devices returned by enumerateDevices() if a wired headset is plugged. I will implement that.

Reply:
Like we talked we are only showing the bluetooth now when It is available. If we connect the wired headset we dont return the bluetooth anymore.